src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -0,0 +1,162 @@
+                new StorageBlobContainerComparer());
+            _pollStrategies = new Dictionary<IStorageBlobContainer, PollStrategy>(
+                new StorageBlobContainerComparer());
+            _scanContainerStrategy = new ScanContainersStrategy();
+            _pollLogStrategy = new PollLogsStrategy();
+            _cancellationTokenSource = new CancellationTokenSource();
+        }
+
+        public void Start()
+        {
+            ThrowIfDisposed();
+            // can not start the 2 strategies till all regiesterations are done
+            ScnaContainersAndRegisterWithStrategy(_cancellationTokenSource.Token);
+            _scanContainerStrategy.Start();
+            _pollLogStrategy.Start();
+        }
+
+        public void Cancel()
+        {
+            ThrowIfDisposed();
+            _scanContainerStrategy.Cancel();
+            _pollLogStrategy.Cancel();
+            _cancellationTokenSource.Cancel();
+        }
+
+        public Task RegisterAsync(IStorageBlobContainer container, ITriggerExecutor<IStorageBlob> triggerExecutor, CancellationToken cancellationToken)
+        {
+            ThrowIfDisposed();
+            // Register and Execute are not concurrency-safe.
+            // Avoiding calling Register while Execute is running is the caller's responsibility.
+
+            ICollection<ITriggerExecutor<IStorageBlob>> containerRegistrations;
+            if (_registrations.ContainsKey(container))
+            {
+                containerRegistrations = _registrations[container];
+            }
+            else
+            {
+                containerRegistrations = new List<ITriggerExecutor<IStorageBlob>>();
+                _registrations.Add(container, containerRegistrations);
+            }
+
+            containerRegistrations.Add(triggerExecutor);
+            return Task.FromResult(0);
+        }
+
+        public async Task<TaskSeriesCommandResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            ThrowIfDisposed();
+            await Task.WhenAll(_scanContainerStrategy.ExecuteAsync(cancellationToken),
+                                _pollLogStrategy.ExecuteAsync(cancellationToken));
+
+            // Run subsequent iterations at 2 second intervals.
+            return new TaskSeriesCommandResult(wait: Task.Delay(TwoSeconds));
+        }
+
+        public void Notify(IStorageBlob blobWritten)
+        {
+            ThrowIfDisposed();
+            _scanContainerStrategy.Notify(blobWritten);
+            _pollLogStrategy.Notify(blobWritten);
+        }
+
+        public void Dispose()
+        {
+            if (!_disposed)
+            {
+                _pollLogStrategy.Dispose();
+                _scanContainerStrategy.Dispose();
+                _cancellationTokenSource.Dispose();
+                _disposed = true;
+            }
+        }
+
+        private void ScnaContainersAndRegisterWithStrategy(CancellationToken cancellationToken)
```

Comment:
Recommend simply renaming this method to **Initialize** since it's job is to perform any pre-Start initialization required. I find the current name hard to understand.